mule-module-apikit/src/main/java/org/mule/module/apikit/AbstractRouter.java

```diff
@@ -67,6 +67,24 @@ public void setFlowConstruct(FlowConstruct flowConstruct)
         this.flowConstruct = flowConstruct;
     }
 
+    @Override
+    public final MuleEvent process(MuleEvent event) throws MuleException
+    {
+        if (isNonBlocking(event))
+        {
+            return processNonBlocking(event);
+        }
+        else
+        {
+            return processBlocking(event);
+        }
+    }
+
+    private boolean isNonBlocking(MuleEvent event)
+    {
+        return event.isAllowNonBlocking() && event.getReplyToHandler() != null;
+    }
+
     protected MuleEvent processBlocking(MuleEvent event) throws MuleException
     {
         if (config.isExtensionEnabled() && config.getRouterExtension().isExecutable(event))
```

Comment:
I have a doubt here. If the event is non blocking, then the extension is not called. Are we ok with this behaviour?